src/content/en/updates/2018/07/reportingobserver.md

```diff
@@ -0,0 +1,238 @@
+because I found it fascinating how many APIs there are for monitoring the
+"stuff" that happens in a web app. For example, there are APIs that can observe
+information about the DOM: `ResizeObserver`,
+`IntersectionObserver`, `MutationObserver`. There are APIs for capturing
+performance measurements: `PerformanceObserver`. Other
+APIs like `window.onerror` and `window.onunhandledrejection` even let us know
+when something goes wrong.
+
+However, there are other types of warnings which are not captured by these
+existing APIs. When your site uses a deprecated API or runs up
+against a [browser intervention][interventions], DevTools is first to tell you
+about them:
+
+<figure>
+  <img src="/web/updates/images/2018/07/reporting/consolewarnings.png"
+       class="screenshot" alt="DevTools console warnings for deprecations and interventions."
+       title="DevTools console warnings for deprecations and interventions.">
+  <figcaption>Browser-initiated warnings in the DevTools console.</figcaption>
+</figure>
+
+One would naturally think `window.onerror` captures these warnings. It does not!
+That's because `window.onerror` does not fire for warnings
+generated directly by the user agent itself. It fires for runtime errors
+(JS exceptions and syntax errors) caused by executing your code.
+
+`ReportingObserver` picks up the slack. It provides a programmatic way to be
+notified about browser-issued warnings such as [deprecations][deprecations]
+and [interventions][interventions]. You can use it as a reporting tool and
+lose less sleep wondering if users are hitting unexpected issues on your live
+site.
+
+`ReportingObserver` is part of a larger spec, the [Reporting API][spec],
+which provides a common way to send these different reports to a backend.
+The Reporting API is basically a generic framework to specify a set of server
+endpoints to report issues to.
+{: .key-point }
+
+## The API {: #api }
+
+The API is not unlike the other "observer" APIs such
+as `IntersectionObserver` and `ResizeObserver`. You give it a callback;
+it gives you information. The information that the callback receives is a
+list of issues that the page caused:
+
+```js
+const observer = new ReportingObserver((reports, observer) => {
+  for (const report of reports) {
+    // → report.id === 'XMLHttpRequestSynchronousInNonWorkerOutsideBeforeUnload'
+    // → report.type === 'deprecation'
+    // → report.url === 'https://reporting-observer-api-demo.glitch.me'
+    // → report.body.message === 'Synchronous XMLHttpRequest is deprecated...'
+    // → report.body.lineNumber === 11
+    // → report.body.columnNumber === 22
+    // → report.body.sourceFile === 'https://reporting-observer-api-demo.glitch.me'
+    // → report.body.anticipatedRemoval === <JS_DATE_STR> or null
+  }
+}});
+
+observer.observe();
+```
+
+#### Filtered reports {: #filtered }
+
+Reports can be pre-filter to only observe certain report types:
```

Comment:
'pre-filtered'?

Also, are the reports filtered or the observation not done for types not included?